basic_subscriber.go

```diff
@@ -79,10 +81,43 @@ func (s *BasicSubscriber) Subscribe(ctx context.Context, gotData func(seqNo uint
 		glog.V(5).Infof("New peer from kademlia: %v", peer.IDHexEncode(p))
 		ns := s.Network.NetworkNode.GetOutStream(p)
 		if ns != nil {
```

Comment:
Is `Subscribe` the right place to send `TranscodeSub`?  Nodes that are not transcoders can also try and subscribe to the video, and I don't think they should be able to get a direct connection with the broadcaster.

What do you think about adding `TranscodeSub(StrmID)` to the `VideoNetwork` interface, so the node will only call it when it's a transcoder?  

I also think this message should be an enhancement rather than a requirement - meaning if for some reason the broadcaster cannot create a direct connection with the transcoder, everything should still work in the relayer-based model.

Reply:
You're right that is probably a better approach. This setup was based on a brief discussion we had in gitter, which I likely mis-interpreted.

Agreed on the relay based fallback; that is on the to-do list ('Resilience' bullet point in https://github.com/livepeer/go-livepeer-basicnet/issues/29). In order to do that, the broadcaster needs to nack the TranscodeSub request, so we need an additional message.

Reply:
Fixed in https://github.com/livepeer/go-livepeer-basicnet/pull/32/commits/242fee8cd3601bdddb87888a1cb553814fcedfb5